src/gui/categoryfilterproxymodel.cpp

```diff
@@ -50,8 +50,11 @@ bool CategoryFilterProxyModel::lessThan(const QModelIndex &left, const QModelInd
 {
     // "All" and "Uncategorized" must be left in place
     if (CategoryFilterModel::isSpecialItem(left) || CategoryFilterModel::isSpecialItem(right))
-        return left.row() < right.row();
-    else
-        return Utils::String::naturalCompareCaseInsensitive(
-                left.data().toString(), right.data().toString());
+        return left < right;
```

Comment:
Why did you change it from comparing row number?

Reply:
What is happening when we compare two QModelIndex's?

Reply:
>What is happening when we compare two QModelIndex's?

qt doc isn't helpful, so here's the source:
https://github.com/qt/qtbase/blob/3fe74b76fd0eaf39d4c6681e2edca5adbf107883/src/corelib/itemmodels/qabstractitemmodel.h#L77

it compares the row then the column then the internal addresses, I think its pretty safe to use it here.